src/units/types.cpp

```diff
@@ -1367,7 +1367,9 @@ void unit_type::apply_scenario_fix(const config& cfg)
 	}
 	if(auto attr = cfg.get("add_advancement")) {
 		for(const auto& str : utils::split(attr->str())) {
-			advances_to_.push_back(str);
+			if(!utils::contains(advances_to_, str)) {
```

Comment:
Honestly would probably be better to make `advances_to_` a `set`.

Reply:
Probably, but that is not something to ask during this pr.

Reply:
Iirc the order is important though, since off-turn advancements always choose the first one. So changing it to set might change behavior

Reply:
Having it always choose the first advancement when a unit advances off-turn is intentional.

Reply:
Right, but making it be a set means a designer can't control which is the default advancement in this case – it'll take the one that's alphabetically first. It's better to leave open that avenue of designer control, I think. So no set, just an array which is kept unique.

Reply:
> Having it always choose the first advancement when a unit advances off-turn is intentional.

Well yes, did something i said make it sound like that's not the case?